### ISSUE TITLE
Column `slug` does not exists

### ISSUE THREAD

**Issue by ppalmeida** [rinvex/subscriptions#23]
Hi, guys.

When I try to assert the use of a feature, I do like this:

```php
// (... subscription is made on Controller, it goes ok as far as I can see)... Then:

$plan = app('rinvex.subscriptions.plan')->find(1);
$subscriptions = $plan->subscriptions;
$user->refresh();
$this->assertTrue($user->subscription('pro')->canUseFeature('news'));
```

And the error in console is:
```
1) Tests\Feature\SusbscriptionTest::a_user_can_subscribe_to_plan
Illuminate\Database\QueryException: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'slug' in 'where clause' (SQL: select * from `plan_subscriptions` where `plan_subscriptions`.`user_id` = 1 and `plan_subscriptions`.`user_id` is not null and `plan_subscriptions`.`user_type` = App\User and `slug` = pro limit 1)
...
Caused by
PDOException: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'slug' in 'where clause'
```
I only could find `slug` in code at `Rinvex\Subscriptions\Traits\HasSubscriptions`:

```php
public function subscription(string $subscriptionSlug): ?PlanSubscription
    {
        return $this->subscriptions()->where('slug', $subscriptionSlug)->first();
    }
```

If I change that 'slug' to 'name', I get this error:

```
1) Tests\Feature\SusbscriptionTest::a_user_can_subscribe_to_plan
Error: Call to a member function expired() on null

/Users/pedropauloalmeida/WORK/PROJECTS/WAVEWEB/SPACE-ROCKET/space-rocket/vendor/rinvex/subscriptions/src/Models/PlanSubscription.php:510
/Users/pedropauloalmeida/WORK/PROJECTS/WAVEWEB/SPACE-ROCKET/space-rocket/tests/Feature/SusbscriptionTest.php:33
```
So I checked the $plan->features->pluck('name') and the result is as expected:

```php
Illuminate\Support\Collection {#704
  #items: array:4 [
    0 => "news"
    1 => "communications"
    2 => "help"
    3 => "library"
  ]
}
```

So, the problems I've found are:
1) The  `Rinvex\Subscriptions\Traits\HasSubscriptions` uses "slug" column, that do not exists anymore.
2) If I change that column to "name", the model `/Models/PlanSubscription.php:510` leads to a null point exception.

Could you please check it out?

**Oh, a last warn: could it be related to this issue?
https://github.com/rinvex/subscriptions/issues/22**

Thank you.

**Comment by ppalmeida** [comment]
One more thing:

In `PlanSubscription` model, the problem is in ```$usage``` variable, line 502: it returns NULL (that's the null point):
```$usage = $this->usage()->byFeatureName($featureName)->first();```

The query was scoped correctly (I think):
```sql
select * from `plan_subscription_usage` where `plan_subscription_usage`.`subscription_id` = ? and `plan_subscription_usage`.`subscription_id` is not null and `feature_id` = ?
```
But at line 510, the first attempt is ```if ($usage->expired()``` on a NULL pointer.

Hope it helps.

https://github.com/rinvex/subscriptions/blob/285f08eaec97048b4fec58da51ef8c083fdf5847/src/Models/PlanSubscription.php#L499

**Comment by Omranic** [comment]
Fixed in 96dad848fbc3d1aeb1a367231dc651b16749bf12 Please feel free to open this or a new issue if you need further assistance 👍
